clang/include/clang/Basic/AttributeCommonInfo.h

```diff
@@ -191,6 +191,12 @@ class AttributeCommonInfo {
   /// __gnu__::__attr__ will be normalized to gnu::attr).
   std::string getNormalizedFullName() const;
 
+  /// Gets a normalized full name, with syntax, scope and name.
+  static std::string
+  getNormalizedFullNameWithSyntax(const IdentifierInfo *Name,
+                                  const IdentifierInfo *Scope,
+                                  AttributeCommonInfo::Syntax SyntaxUsed);
```

Comment:
It feels inconsistent to have a member function `getX` and a static member `getXWithY`. Might be better to spell this with `normalize` as the verb, like `normalizeFullNameWithSyntax`